### PR TITLE
Feature/select

### LINKED ISSUE #16: [UI core library] Select
https://zeroheight.com/11d0e6dac/p/12bbb4-dropdown

Implement only the classic dropdown menu initially

1. Create a library-agnostic component
    1. API (extend https://formik.org/docs/api/useField#fieldinputpropsvalue)
        1. name
        2. value
        3. onChange
        4. onBlur
        5. label
        6. error
        7. touched
        8. options (as children)
2. Create a formik v2 wrapper

### PR DESCRIPTION
**Pull request by jankoritak**
Closes #16

### CODE REVIEW

File: packages/ui/src/Select.tsx
Comment (aigoncharov) [inline packages/ui/src/Select.tsx:52]: nit: I would suggest something language-neutral like `-`

File: packages/ui/src/Select.tsx
Comment (aigoncharov) [inline packages/ui/src/Select.tsx:71]: If we use native `select` than our `Select` is going to look quite differently compared to `Multiselect`. Should we use some library that can cover both cases for consistent look and feel?

Comment (aigoncharov) [reply packages/ui/src/Select.tsx:71]: I think `react-select` worked out just perfectly for us at MC. I am researching other options now.

Comment (jvorcak) [reply packages/ui/src/Select.tsx:71]: @aigoncharov I agree with using one lib for both scenarios

Comment (jankoritak) [reply packages/ui/src/Select.tsx:71]: Good point! I totally overlooked the fact there will be a multi-select component. As much as I don't like `react-select` styling API, I think we should stick to it. It has proven to be a great match for us.

Comment (jvorcak) [reply packages/ui/src/Select.tsx:71]: @aigoncharov @jankoritak we already use react-select as well on .cloud.

Comment (aigoncharov) [reply packages/ui/src/Select.tsx:71]: @jankoritak @jankoritak there is an alternative to `react-select`. We could use low-level [downshift](https://github.com/downshift-js/downshift) and build a custom select ourselves. We already have `popper` to position the menu, so we'll just need to style it. However, I could imagine it might take a while to do it right.

Comment (aigoncharov) [reply packages/ui/src/Select.tsx:71]: I gave it some thinking and I feel like we would be better off with `reatc-select`. It might be a bit bulky, but we have saved KB with other components, right? Time is of the matter, we should start integration ASAP. We could always reimplement the component matching the existing API later on when we have time. Let's do simple stuff for now. WDYT?

Comment (aigoncharov) [reply packages/ui/src/Select.tsx:71]: @jankoritak could you prioritize this task over your other ones? I think `Select` and `Password` are only ones left to start with the new login flow for MC.

Comment (jankoritak) [reply packages/ui/src/Select.tsx:71]: Certainly, I'll start with this right away!

Comment (danad02) [reply packages/ui/src/Select.tsx:71]: Implementing the components(s) using downshift would take a lot of time. I would stick to `react-select`, at least for now.

File: package.json
Comment (aigoncharov) [inline package.json:15]: We will move this types to `@hazelcast/ui` later in scope of https://github.com/hazelcast/frontend-shared/issues/59

File: packages/ui/src/SelectField.tsx
Comment (danad02) [inline packages/ui/src/SelectField.tsx:109]:
```suggestion
        // Menu container is either this select itself or document.body
```

File: packages/ui/src/SelectField.tsx
Comment (danad02) [inline packages/ui/src/SelectField.tsx:129]: I would love to see multi select and searchable select stories.

File: packages/ui/src/SelectField.tsx
Comment (danad02) [inline packages/ui/src/SelectField.tsx:109]: > Menu container is either this select itself or document.body
According to `getMenuContainer` it could be any element. Or am I mixing two different things here?

File: packages/ui/src/SelectField.module.scss
Comment (danad02) [inline packages/ui/src/SelectField.module.scss:68]: I can't quite get what's going on here from the message :cry: .

File: packages/ui/__tests__/SelectField.test.tsx
Comment (danad02) [inline packages/ui/__tests__/SelectField.test.tsx:3]:
```suggestion
```

File: packages/ui/__tests__/SelectFormik.test.tsx
Comment (danad02) [inline packages/ui/__tests__/SelectFormik.test.tsx:1]: nit: Can we rename the file to `SelectFieldFormik.test.tsx` for consistency?

**Review by danad02 — COMMENTED** [review]
@aigoncharov Left several comments.

File: packages/ui/src/SelectField.tsx
Comment (aigoncharov) [reply packages/ui/src/SelectField.tsx:129]: Totally forgot about searchable select. Thnx. As to multiselect. I think it should be a different component.

Comment (aigoncharov) [reply packages/ui/src/SelectField.tsx:129]: Addressed in https://github.com/hazelcast/frontend-shared/pull/72/commits/acbad02b1ebd440112f7c2ef53b08a90a6b49557

File: packages/ui/src/SelectField.tsx
Comment (jankoritak) [inline packages/ui/src/SelectField.tsx:58]: Nit: We can pick `required` and `placeholder` from `InputHTMLAttributes` as well.

```suggestion
export type SelectFieldExtraProps<V> = {
  options: SelectFieldOption<V>[]
  label: string
  helperText?: string | ReactElement
  className?: string
  errorClassName?: string
  menuPortalTarget?: 'body' | 'self' | HTMLElement | null
} & DataTestProp &
  Pick<InputHTMLAttributes<HTMLElement>, 'autoFocus' | 'disabled' | 'autoComplete' | 'placeholder' | 'required'> &
  Pick<ReactSelectProps, 'isSearchable' | 'menuIsOpen' | 'menuPlacement' | 'noOptionsMessage' | 'inputValue'>
```

File: packages/ui/src/SelectField.tsx
Comment (danad02) [reply packages/ui/src/SelectField.tsx:129]: @aigoncharov So we're going to use `SelectField` as a base for `MultiSelectField` component, right?

**Review by jankoritak — COMMENTED** [review]
Stellar work! Cannot approve due to being author of the PR. Leaving that up to @danad02.

**Review by danad02 — APPROVED** [review]
Great job! Ready to merge once we settle on https://github.com/hazelcast/frontend-shared/pull/72#discussion_r535131045

File: packages/ui/src/SelectField.tsx
Comment (aigoncharov) [reply packages/ui/src/SelectField.tsx:127]: @danad02 currently, `SelectField` does not accept `isMulti` as a prop, so you won't be able to do it unless you change the code :) I just feel like MultiSelect has a completely different look and feel, so I would rather do it as a standalone component.

Comment (danad02) [reply packages/ui/src/SelectField.tsx:127]: Ah got it :smile: Well there is no information about it in the style guide. But if the future designs tell us to use this as a multiselect then there is nothing easier than enabling it and change styles a bit. Approved!

Comment (aigoncharov) [reply packages/ui/src/SelectField.tsx:127]: @danad02 I think we will implement MultiSelect based on the same react-select. I just do not want to do with lots of conditional code to make it happen. I would rather copy-paste something from this component into a new shiny `MultiSelect` and make it works there for the multi select use-case only

Comment (danad02) [reply packages/ui/src/SelectField.tsx:127]: Yeah there would probably be a surprising number of conditions :+1:

**Pull request merged**
